src/temporal-logic/ctl.cpp

```diff
@@ -0,0 +1,16 @@
+/*******************************************************************\
+
+Module: CTL
+
+Author: Daniel Kroening, dkr@amazon.com
+
+\*******************************************************************/
+
+#include "ctl.h"
+
+#include "temporal_logic.h"
+
+bool is_AGp(const exprt &expr)
+{
+  return expr.id() == ID_AG && !has_temporal_operator(to_AG_expr(expr).op());
+}
```

Comment:
Surprisingly, this is different from `is_AGp` in `bdd_enginet::check_property`?!

Reply:
Now clearer, avoiding labeling expressions as AGp that are just equivalent.